test/features/events.js

```diff
@@ -0,0 +1,312 @@
+
+      Event.find({
+        where: {
+          timestamp: dateString,
+          operation: 'check in',
+          centre: 1
+        }
+      }).then((models) => expect(models.length).to.equal(1))
+    );
+
+  });
+
+  Scenario('Event should be created', () => {
+
+    var date = new Date();
+    var dateString = date.toISOString();
+    var payload = {
+      person_id: 123,
+      cid_id: 100,
+      operation: 'check in',
+      timestamp: dateString,
+      gender: 'm',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    var createEvent = function () {
+      return request(sails.hooks.http.app)
+        .post('/irc_entry/event')
+        .send(payload)
+        .expect(201);
+    }
+
+    Given('a detainee event with timestamp "' + dateString + '" has already occurred', () =>
+
+      global.initializeBarrelsFixtures().then(() =>
+        createEvent().then(() =>
+          Event.find({
+            where: {
+              timestamp: dateString,
+              operation: 'check in'
+            }
+          }).then((models) => expect(models.length).to.equal(1))
+        )
+      )
+    );
+
+    When('a valid detainee event with timestamp "' + dateString + '" occurs', createEvent);
+
+    Then('a detainee event with timestamp "' + dateString + '" should be created from the check in received', () =>
+
+      Event.find({
+        where: {
+          timestamp: dateString,
+          operation: 'check in'
+        }
+      }).then((models) =>
+        expect(models.length).to.equal(2)
+      )
+    );
+
+  });
+
+  Scenario('Existing Detainee Should Be Updated From Check In Operation That Occurred After The Detainee Creation', () => {
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    var eventUpdateDateString = (new Date(20000)).toISOString();
+    var detaineeUpdateDateString = (new Date(10000)).toISOString();
+
+    var payload = {
+      person_id: 123,
+      cid_id: 4321,
+      operation: 'check in',
+      timestamp: eventUpdateDateString,
+      gender: 'f',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    var detaineeAttrs = {
+      person_id: 123,
+      cid_id: 1234,
+      centre: 1,
+      gender: 'male',
+      operation: 'check in',
+      timestamp: detaineeUpdateDateString
+    }
+
+    var getSearchResult = () => Detainee.find({
+      centre: detaineeAttrs.centre,
+      person_id: detaineeAttrs.person_id
+    });
+
+    Given('a Detainee already exists with person id `' + detaineeAttrs.person_id + '` from centre `' + detaineeAttrs.centre + '`', () =>
```

Comment:
this could be made more readable with es6 template syntax:
`Given(`a Detainee already exists with person id '${detaineeAttrs.person_id}' from centre ${detaineeAttrs.centre}'`, () => `
rather than string concat